mii/utils.py

```diff
@@ -110,17 +110,23 @@ def check_if_task_and_model_is_valid(task, model_name):
     ), f"{task_name} only supports {valid_task_models}"
 
 
-def aml_model_path(model_path):
-    aml_model_dir = os.getenv("AZUREML_MODEL_DIR")
-
-    # AML local deployments this will be None
-    if aml_model_dir is None and model_path is None:
-        return mii.constants.MII_MODEL_PATH_DEFAULT
-
-    if model_path:
-        return os.path.join(aml_model_dir, model_path)
+def full_model_path(model_path):
+    aml_model_dir = os.environ.get('AZUREML_MODEL_DIR', None)
+
+    if aml_model_dir:
+        # (potentially) append relative model_path w. aml path
+        assert os.path.isabs(aml_model_dir), f"AZUREML_MODEL_DIR={aml_model_dir} must be an absolute path"
+        if model_path:
+            assert not os.path.isabs(model_path), f"model_path={model_path} must be relative to append w. AML path"
+            return os.path.join(model_path, aml_model_dir)
```

Comment:
```suggestion
            return os.path.join(aml_model_dir, model_path)
```

The `model_path` should be appended, not preprended - right?

Reply:
Ohh great catch, yes you are right.